### PR TITLE
Export as preset instead of plugin

### LINKED ISSUE #2: Remove `jscs` dependency
`jscs` is currently required as a dependency because it's being used for testing the custom rules and validating the additional code written for that matter, and also to allow the rules to be declared in `yaml`.

To shave some bytes from `production` installs, it can be moved to `devDependencies`, which still allows to test the custom rules but prevents the other use-cases.

### PR DESCRIPTION
**Pull request by ruimarinho**
- Removed babel as only tests could potentially benefit from transpiling.
- Cleaned up tests.
- Tidied up package.json.

By the way, it looks like even thought `jscs-config-*` is supported, the jscs community is leaning towards `jscs-preset-*`.

Might help with #8.

Closes #2.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,17 +28,10 @@
...
-    "jscs": "^2.5.0"
-  },
   "devDependencies": {
-    "babel": "^5.8.23",
+    "jscs": "^2.8.0",
     "mocha": "^2.3.3",
-    "should": "^7.1.0"
-  },
-  "peerDependencies": {
-    "jscs": ">= 2.5",
```
Comment (ruiquelhas) [inline package.json]: I think `jscs` should also be included here. I know `peerDependencies` are controversial, but the user must get warned about this kind of stuff, and this is pretty much useless without `jscs`.

File: src/rules/disallow-generators-in-describe-functions.js [hunk 2]
```diff
@@ -4,15 +4,16 @@
...
 /**
  * Export `disallowGeneratorsInDescribeFunctions`.
  */
 
-const disallowGeneratorsInDescribeFunctions = () => {};
+module.exports = function() {
+};
 
-disallowGeneratorsInDescribeFunctions.prototype = {
+module.exports.prototype = {
```
Comment (ruiquelhas) [inline src/rules/disallow-generators-in-describe-functions.js]: Not sure if I like this. I would still keep the named `function`, do stuff on its prototype and then export it.

File: package.json [hunk 3]
```diff
@@ -2,15 +2,13 @@
   "name": "jscs-config-seegno",
```
Comment (ruiquelhas) [inline package.json]: Maybe we should change the package name to `jscs-preset-seegno`.

File: src/rules/disallow-generators-in-describe-functions.js (hunk 2 above)
Comment (ruimarinho) [inline src/rules/disallow-generators-in-describe-functions.js]: While I agree that this looks very weird, every single rule is designed like on the official package. E.g. https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/require-capitalized-constructors.js#L38

There are plans to [migrate the codebase to ES2015](https://github.com/jscs-dev/node-jscs/issues/1469), which implies rewriting all the rules, but I'm still open to change this now if you prefer.

File: package.json (hunk 1 above)
Comment (ruimarinho) [inline package.json]: I agree with you on principle, but the fact that npm@3 still handles peerDependencies in a very disturbing way makes me prefer not keeping this here as I don't think anyone will be using a jscs preset without jscs installed. Not even the owner of the project does that - https://github.com/mrjoelkemp/jscs-preset-mrjoelkemp/blob/master/package.json.

File: package.json (hunk 3 above)
Comment (ruimarinho) [inline package.json]: I'm +1 on that, even though jscs supports all https://github.com/jscs-dev/node-jscs/blob/2481d632053dc0d5ca8020741e0bae4b14d818ed/lib/config/node-configuration.js#L113

File: src/rules/disallow-generators-in-describe-functions.js (hunk 2 above)
Comment (ruiquelhas) [inline src/rules/disallow-generators-in-describe-functions.js]: :+1:

File: package.json (hunk 3 above)
Comment (ruiquelhas) [inline package.json]: I also think it's better.

File: package.json (hunk 1 above)
Comment (ruiquelhas) [inline package.json]: I guess we can still keep it as part of the installation command at least. Let's move on.

**Comment by ruimarinho** [comment]
Can this be merged @ruiquelhas?

**Comment by ruimarinho** [comment]
ping @ruiquelhas

**Pull request merged**